dbms/src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -87,6 +89,33 @@ struct BinaryOperationImplBase
     }
 };
 
+template <typename Op>
+struct FixedStringOperationImpl
+{
+    static void NO_INLINE vector_vector(const ColumnFixedString::Chars & a, const ColumnFixedString::Chars & b, ColumnFixedString::Chars & c)
+    {
+        size_t size = a.size();
+        for (size_t i = 0; i < size; ++i)
+            c[i] = Op::template apply<UInt8>(a[i], b[i]);
+    }
+
+    static void NO_INLINE vector_constant(const ColumnFixedString::Chars & a, const ColumnFixedString::Chars & b, ColumnFixedString::Chars & c)
+    {
+        size_t size = a.size();
+        for (size_t i = 0; i < size; ++i)
+            c[i] = Op::template apply<UInt8>(a[i], b[i % b.size()]);
```

Comment:
Could you please also add performance test?
This code looks suspicious.

Reply:
performance test added